user_agreement_core_lib/data_layers/data_access/user_agreement_document_data_access.py

```diff
@@ -40,25 +64,30 @@ def get_agreed_document_by_name(self, user_id: int, document_name: str, language
             return (
                 session.query(UserAgreementDocument)
                 .join(AgreementDocument)
-                .filter(UserAgreementDocument.user_id == user_id,
-                        AgreementDocument.name == document_name,
-                        AgreementDocument.language == language,
-                        UserAgreementDocument.deleted_at == None)
+                .filter(
+                    UserAgreementDocument.user_id == user_id,
+                    AgreementDocument.name == document_name,
+                    AgreementDocument.language == language,
+                    UserAgreementDocument.deleted_at == None
+                )
                 .order_by(desc(AgreementDocument.version))
                 .first()
             )
 
+
     def delete(self, user_id: int, document_id: int):
         with self.db_session.get() as session:
-            return (
+            updated_count = (
                 session.query(UserAgreementDocument)
-                .filter(UserAgreementDocument.agreement_document_id == document_id,
-                        UserAgreementDocument.user_id == user_id,
-                        UserAgreementDocument.deleted_at == None)
-                .update(
-                    {
-                        UserAgreementDocument.deleted_at: datetime.utcnow(),
-                        UserAgreementDocument.deleted_at_token: int(datetime.utcnow().timestamp()),
-                    }
+                .filter(
+                    UserAgreementDocument.agreement_document_id == document_id,
+                    UserAgreementDocument.user_id == user_id,
+                    UserAgreementDocument.deleted_at == None
                 )
+                .update({
+                    UserAgreementDocument.deleted_at: datetime.utcnow(),
+                    UserAgreementDocument.deleted_at_token: int(datetime.utcnow().timestamp()),
+                })
             )
+            session.commit()
+            return updated_count > 0
```

Comment:
Thank you Oren. for a bit of order: 

- no need fot the session commit here. the with context manager should handle that. 
- just return directly. let the service do the if condition
what do you thing?